src/include/platform/DiagnosticDataProvider.h

```diff
@@ -70,7 +70,7 @@ class GeneralDiagnosticsDelegate
      * @brief
      *   Called after the current device is rebooted.
      */
-    virtual void OnDeviceRebooted() {}
+    virtual void OnDeviceRebooted(uint8_t bootReason) {}
```

Comment:
If this does not use BootReasonType, at bare minimum need to document that the values need to match those values.